Ember New Lang (WIP)

hergaiety opened the pull request:
[Rendered](https://github.com/hergaiety/rfcs/blob/ember-new-lang/text/0000-ember-new-lang.md)
[Candidate Implementation](https://github.com/josephdsumner/ember-cli/compare/master...ember-new-lang-base)

Opened RFC: https://github.com/emberjs/rfcs/pull/635

Commit:
First pre-draft commit to begin collaboration of ember-new-lang RFC

hergaiety committed:
Added more information to Motivation section

hergaiety committed:
Corrected link to issue

hergaiety committed:
More motivation bullet points

hergaiety committed:
Filled out How we teach this

hergaiety committed:
Removed template prompts from Motivation section

hergaiety committed:
Added an unresolved question

hergaiety committed:
Added a drawback

hergaiety committed:
Classic "do nothing" alternative for now

jgwhite commented on text/0000-ember-new-lang.md:
```diff
@@ -0,0 +1,57 @@
+
+## Motivation
+
+This will in part address item 4 within issue https://github.com/emberjs/rfcs/issues/595
```
We need to find a concise way to say here that this RFC does *not* intend to assume a default language, rather its aim is to lay the groundwork for a future interactive solution that prompts every ember user to start out of the gate with a language specified.

hergaiety replied:
Pushed up e66ba23, let me know what you think and feel free to make any adjustments directly you wish.

hergaiety replied:
(hah, typo fix in 61d6e3d)

Commit:
Further defines the motivation GO and NO GO's

Commit:
Fix motivation typo

jgwhite commented on text/0000-ember-new-lang.md:
```diff
@@ -0,0 +1,57 @@
+
+## Motivation
+
+This RFC offers a partial resolution to [this related issue's issue #4](https://github.com/emberjs/rfcs/issues/595). This RFC explicitly does _not_ aim to add a default language, but instead aims to add tooling that offers users a way to specify a language.
```
```suggestion
This RFC offers a partial resolution to technical accessibility issue 4 “Missing default language declaration” from the Ember A11y Strike Team’s [“Technical Accessibility Issues for New Ember Apps” list](https://github.com/emberjs/rfcs/issues/595). This RFC explicitly does _not_ aim to add a default language, but instead aims to add tooling that offers users a way to specify a language.
```

jgwhite commented on text/0000-ember-new-lang.md:
```diff
@@ -0,0 +1,57 @@
+
+## How we teach this
+
+The [Ember CLI API documentation](https://ember-cli.com/api/) should be updated to reflect the new flag.
+
+We update the [Ember.js CLI Guides](https://cli.emberjs.com/release/basic-use/cli-commands/) to reflect the new flag much like we demonstrate `--yarn` usage.
+
+The Ember CLI `--help` command should explain what kind of value is expected to be passed to the new flag.
```
Shall we format this section as an ordered list?

i.e.

```suggestion
1. Update the [Ember CLI API documentation](https://ember-cli.com/api/) to reflect the new flag.
2. Update the [Ember.js CLI Guides](https://cli.emberjs.com/release/basic-use/cli-commands/) to reflect the new flag much like we demonstrate `--yarn` usage.
3. Update the Ember CLI `--help` command so it explains what kind of value is expected to be passed to the `--lang` flag.
```

jgwhite commented on text/0000-ember-new-lang.md:
```diff
@@ -0,0 +1,57 @@
+
+## Drawbacks
+
+More flags means more combinations of ways to run `ember new` which can be hard to test for and is potentially unsustainable.
```
```suggestion
More flags means more combinations of ways to run `ember new` which can be hard to test for and is potentially unsustainable.

Users may be confused about whether they’re supposed to specify a human language or a programming language (i.e. `--language typescript`).
```

hergaiety committed:
Update text/0000-ember-new-lang.md

Co-authored-by: Jamie White <jameswhite@salesforce.com>

hergaiety committed:
Update text/0000-ember-new-lang.md

Co-authored-by: Jamie White <jameswhite@salesforce.com>

hergaiety committed:
Update text/0000-ember-new-lang.md

Co-authored-by: Jamie White <jameswhite@salesforce.com>

Commit:
Unordered list drawbacks

Commit:
Detailed design

Commit:
Added some text before the help command

Commit:
Cleaned up common misunderstandings copy

hergaiety committed:
Converted from --language to --lang

hergaiety committed:
Trimmed help snippet

hergaiety committed:
Updated candidate link

hergaiety committed:
Added note for why we're chosing `--lang`

hergaiety committed:
Notes For and Against setting default lang: en-US

josephdsumner committed:
Update `Summary` section per review

josephdsumner committed:
Fix typo

josephdsumner committed:
Phrasing

josephdsumner committed:
Update `Motivation` section per review

josephdsumner committed:
Add reference to `lang` definition in HTML spec per review

josephdsumner committed:
Update text/0000-ember-new-lang.md per review

Co-authored-by: Melanie Sumner <melaniersumner@gmail.com>

josephdsumner committed:
Update text/0000-ember-new-lang.md per review

Co-authored-by: Melanie Sumner <melaniersumner@gmail.com>

MelSumner committed:
Merge pull request #3 from sharpshark28/ember-new-lang-jds-motivation

Update `Motivation` section per review

MelSumner committed:
Merge pull request #2 from sharpshark28/ember-new-lang-jds-summary

Update `Summary` section per review

MelSumner committed:
Update 0000-ember-new-lang.md

MelSumner committed:
Update 0000-ember-new-lang.md

Updating RFC draft per strike team meeting review

MelSumner committed:
Update text/0000-ember-new-lang.md

MelSumner committed:
Update text/0000-ember-new-lang.md

MelSumner committed:
Merge pull request #4 from sharpshark28/MelSumner-patch-1

Update 0000-ember-new-lang.md

MelSumner committed:
Update 0000-ember-new-lang.md

MelSumner committed:
Update 0000-ember-new-lang.md

- added authors (since there are multiple)
- added information based on the feedback from the [strike team's review](https://github.com/ember-a11y/core-notes/blob/ember-a11y/ember-a11y/2020-05/may-27.md)

MelSumner committed:
Update text/0000-ember-new-lang.md

MelSumner committed:
Merge pull request #5 from sharpshark28/MelSumner-patch-2

Update 0000-ember-new-lang.md

MelSumner committed:
Update 0000-ember-new-lang.md

updating for clarity (no wording changed)

MelSumner committed:
Update 0000-ember-new-lang.md

finishing a sentence that didn't end.

MelSumner committed:
Update 0000-ember-new-lang.md

Added additional detailed design info for cases where the `--lang` flag has no arguments defined

josephdsumner committed:
Merge pull request #6 from sharpshark28/MelSumner-patch-3

Update 0000-ember-new-lang.md

josephdsumner committed:
Update 0000-ember-new-lang.md

MelSumner committed:
Merge pull request #7 from sharpshark28/josephdsumner-patch-1

Update 0000-ember-new-lang.md

hergaiety committed:
ember-intl prose

jgwhite commented on text/0000-ember-new-lang.md:
```diff
@@ -0,0 +1,239 @@
+
+### As this relates to ember-intl
+
+The popular [localization library ember-cli](https://github.com/ember-intl/ember-intl) does not conflict with the addition of this new Ember CLI flag. The addition of this flag offers some out of the box support where it was previously missing for new Ember apps. It is still recommended that globalized apps leverage `ember-intl`.
```
```suggestion
The popular [localization library ember-intl](https://github.com/ember-intl/ember-intl) does not conflict with the addition of this new Ember CLI flag. The addition of this flag offers some out of the box support where it was previously missing for new Ember apps. It is still recommended that globalized apps leverage `ember-intl`.
```

hergaiety committed:
Update text/0000-ember-new-lang.md

Co-authored-by: Jamie White <jameswhite@salesforce.com>

MelSumner committed:
Update 0000-ember-new-lang.md

MelSumner committed:
Update 0000-ember-new-lang.md

Update to address feedback from @locks

hergaiety commented:
Closing this draft to not have confusion with the open RFC https://github.com/emberjs/rfcs/pull/635

@MelSumner ~~I believe you pushed 3d358bd here and it maybe should be in the other PR. If it's easier for me to move this commit I'd be happy to, just let me know.~~ Oh nvm, the commit went in both places :)
